addons/pos_pine_labs/static/src/app/utils/payment/payment_pine_labs.js

```diff
@@ -0,0 +1,221 @@
+        if (response.errorMessage) {
+            this._showError(response.errorMessage);
+            return false;
+        }
+        this._pineLabsHandleResponse(response);
+        this._removePaymentHandler();
+        return Promise.resolve(true);
+    }
+
+    /**
+     * This method processes order data and sends payment requests from POS to Pine Labs.
+     */
+    async _processPineLabs() {
+        const order = this.pos.get_order();
+        const paymentLine = order.get_selected_paymentline();
+        const sequenceNumber = order.payment_ids.filter(
+            (pi) => pi.payment_method_id.use_payment_terminal === "pine_labs"
+        ).length;
+        if (paymentLine.amount < 0) {
+            this._showError(_t("Cannot process transactions with negative amount."));
+            return false;
+        }
+
+        const orderId = order?.pos_reference?.replace(" ", "").replaceAll("-", "").toUpperCase();
+        const referencePrefix = this.pos.config.name.replace(/\s/g, "").slice(0, 4);
+        paymentLine.update({
+            payment_ref_no:
+                referencePrefix + "/" + orderId + "/" + crypto.randomUUID().replaceAll("-", ""),
+        });
+
+        // Assume that the Pine Labs terminal payment method is configured with INR (Indian Rupees) as the currency_id in the POS config.
+        // The conversion rate between INR and paisa is set as 1 INR = 100 paisa.
+        const data = {
+            amount: paymentLine.amount * 100, // We need to provide the amount in paisa since Pine Labs processes amounts in paisa.
+            transactionNumber: paymentLine.payment_ref_no,
+            sequenceNumber: sequenceNumber, // In the case of multiple transactions for the same order, it is important to follow the correct sequence of transactions.
+        };
+        const response = await this._callPineLabs(data, "pine_labs_make_payment_request");
+        return await this._pineLabsHandleResponse(response);
+    }
+
+    /**
+     * This method waits for the payment to be confirmed by Pine Labs.
+     * Also, this method uses polling to check the payment status..
+     */
+    async _waitForPaymentToConfirm() {
+        const paymentLine = this.pos.get_order().get_selected_paymentline();
+        if (!paymentLine || paymentLine.payment_status == "retry") {
+            return false;
+        }
+        const data = {
+            plutusTransactionReferenceID: paymentLine.pine_labs_plutus_transaction_ref,
+        };
+        this._stopPendingPayment().then(() => (this.payment_stopped = true));
+        const pineLabsFetchPaymentStatus = async (resolve, reject) => {
+            //Clear the previous timeout before setting a new one
+            clearTimeout(this.pollingTimeout);
+
+            // If the user navigates to another screen, stop the polling
+            if (this.pos.mainScreen.component.name !== "PaymentScreen") {
+                return;
```

Comment:
```suggestion
                this. _removePaymentHandler();
                return;
```

I think you should do this.